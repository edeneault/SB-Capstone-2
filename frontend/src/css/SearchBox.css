.smaller-input {
  font-size: 12px;
  padding: 5px 5px;
  height: 30px;
}

.smaller-button {
  margin-right: 5px;
  padding: 5px 5px;
  font-size: 8px;
  height: 34px;
}
