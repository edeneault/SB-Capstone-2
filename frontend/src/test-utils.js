import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
