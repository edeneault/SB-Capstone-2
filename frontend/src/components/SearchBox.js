import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Form, Button, Container, Row, Col } from "react-bootstrap";
import "../css/SearchBox.css";

const SearchBox = ({ name }) => {
  const history = useHistory();
  const [keyword, setKeyword] = useState("");

  const submitHandler = (e) => {
    e.preventDefault();
    if (keyword.trim()) {
      history.push(`/search/${keyword}`);
    } else {
      history.push("/");
    }
  };
  return (
    <>
      <Container>
        <Row className='d-flex py-2 px-3'>
          <Col>
            <Form onSubmit={submitHandler} className='d-flex ms-auto mt-1'>
              <Form.Control
                id='search'
                type='text'
                name='q'
                onChange={(e) => setKeyword(e.target.value)}
                placeholder={`search ${name}`}
                className='rounded-start smaller-input'
                size='sm'
              ></Form.Control>

              <Button
                type='submit'
                variant='secondary '
                className='btn rounded-end smaller-button'
              >
                <i className='fas fa-search'></i>
              </Button>
              {/* </InputGroup> */}
            </Form>
          </Col>
        </Row>
      </Container>
    </>
  );
};

export default SearchBox;
