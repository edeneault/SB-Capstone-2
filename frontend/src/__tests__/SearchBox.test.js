import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState } from "../store-mock";
import { render, screen } from "../test-utils";

import SearchBox from "../components/SearchBox";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  let name;
  beforeEach(() => {
    name = "lavazza";
  });

  test("renders SearchBox component without crashing", () => {
    const store = mockStore(initialState);

    const { getByRole } = render(
      <Provider store={store}>
        <SearchBox name={name} />
      </Provider>,
    );

    expect(screen.getByRole("textbox").placeholder).toEqual("search lavazza");
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <SearchBox name={name} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
