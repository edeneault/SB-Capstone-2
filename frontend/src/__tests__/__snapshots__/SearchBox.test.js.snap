// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="d-flex py-2 px-3 row"
    >
      <div
        class="col"
      >
        <form
          class="d-flex ms-auto mt-1"
        >
          <input
            class="rounded-start smaller-input form-control form-control-sm"
            id="search"
            name="q"
            placeholder="search lavazza"
            type="text"
          />
          <button
            class="btn rounded-end smaller-button btn btn-secondary "
            type="submit"
          >
            <i
              class="fas fa-search"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
